iOS/Airbnb_iOS/Airbnb_iOS/FindAccomodation/Controller/CalendarViewController.swift

```diff
@@ -9,42 +9,21 @@ import UIKit
 
 protocol CalendarViewControllerDelegate: AnyObject {
     func didSetDateRange(_ dateRange: ClosedRange<Date>)
+    func didChangeDateRange()
 }
 
 final class CalendarViewController: UIViewController {
 
     private lazy var calendarView = CalendarView(frame: view.frame)
 
+    private var collectionDataSource: CalendarCollectionDataSource
     private var selectedDate: Date?
-    private var baseDate: Date {
-        didSet {
-            days = (0..<6).map { offset in
-                generateDaysInMonth(for: Calendar.current.date(byAdding: .month, value: offset, to: baseDate) ?? Date())
-            }
-            calendarView.reloadData()
-        }
-    }
-
-    private lazy var days: [[Day]] = {
-        let days = (0..<12).map { offset in
-            generateDaysInMonth(for: Calendar.current.date(byAdding: .month, value: offset, to: baseDate) ?? Date())
-        }
-
-        return days
-    }()
-
-    private var numberOfWeeksInBaseDate: Int {
-        calendar.range(of: .weekOfMonth, in: .month, for: baseDate)?.count ?? 0
-    }
-
-    private let calendar = Calendar(identifier: .gregorian)
-    private var useCase = CalendarViewControllerUseCase()
+    private var useCase = CalendarUseCase()
 
     private var delegate: CalendarViewControllerDelegate?
```

Comment:
CalendarViewController의 delegate 강한 참조가 되면 CalendarViewController 메모리 해제하는 시점에서 delegate에서 레퍼런스 카운트 감소하지 못하고 1로 유지하고 있기 때문에 CalendarViewController 객체가 메모리에서 해제되지 않고 남아있습니다. `delegate retain cycle in swift` 구글 검색하면 관련 내용에 대해서 확인해보시길 바랍니다🤔

```suggestion
    private weak var delegate: CalendarViewControllerDelegate?
```

Reply:
답변 감사합니다! delegate 프로퍼티의 참조 카운팅에 대해서 생각 못 하고 있었네요.. 해당 부분 약한 참조로 변경하도록 하곘습니다!